src/t8_geometry/t8_geometry_helpers.c

```diff
@@ -201,3 +220,223 @@ t8_geom_get_edge_vertices (const t8_eclass_t tree_class,
+    break;
+  }
+}
+
+void
+t8_geom_get_triangle_scaling_factor (int edge_index,
+                                     const double *tree_vertices,
+                                     const double *glob_intersection,
+                                     const double *glob_ref_point,
+                                     double *scaling_factor)
+{
+  double              dist_intersection, dist_ref;
+  /* The scaling factor depents on the relation of the distance of the opposite vertex
+   * to the global reference point and the distance of the opposite vertex to the global
+   * intersection on the edge.
+   */
+  switch (edge_index) {
+  case 0:                      /* edge 0 */
+    /* The distance for both, the intersection and the reference point, are calculated
+     * using the theorem of pythagoras.
+     */
+    dist_intersection =
+      sqrt (pow ((tree_vertices[0] - glob_intersection[0]), 2)
+            + pow ((tree_vertices[1] - glob_intersection[1]), 2)
+            + pow ((tree_vertices[2] - glob_intersection[2]), 2));
+    dist_ref = sqrt (pow ((tree_vertices[0] - glob_ref_point[0]), 2)
+                     + pow ((tree_vertices[1] - glob_ref_point[1]), 2)
+                     + pow ((tree_vertices[2] - glob_ref_point[2]), 2));
+    /* The closer the reference point is to the intersection, the bigger is the scaling factor. */
+    *scaling_factor = dist_ref / dist_intersection;
+    break;
+  case 1:                      /* edge 1 */
+    /* The distance for both, the intersection and the reference point, are calculated
+     * using the theorem of pythagoras.
+     */
+    dist_intersection =
+      sqrt (pow ((tree_vertices[3] - glob_intersection[0]), 2)
+            + pow ((tree_vertices[4] - glob_intersection[1]), 2)
+            + pow ((tree_vertices[5] - glob_intersection[2]), 2));
+    dist_ref = sqrt (pow ((tree_vertices[3] - glob_ref_point[0]), 2)
+                     + pow ((tree_vertices[4] - glob_ref_point[1]), 2)
+                     + pow ((tree_vertices[5] - glob_ref_point[2]), 2));
+    /* The closer the reference point is to the intersection, the bigger is the scaling factor. */
+    *scaling_factor = dist_ref / dist_intersection;
+    break;
+  case 2:                      /* edge 2 */
+    /* The distance for both, the intersection and the reference point, are calculated
+     * using the theorem of pythagoras.
+     */
+    dist_intersection =
+      sqrt (pow ((tree_vertices[6] - glob_intersection[0]), 2)
+            + pow ((tree_vertices[7] - glob_intersection[1]), 2)
+            + pow ((tree_vertices[8] - glob_intersection[2]), 2));
+    dist_ref = sqrt (pow ((tree_vertices[6] - glob_ref_point[0]), 2)
+                     + pow ((tree_vertices[7] - glob_ref_point[1]), 2)
+                     + pow ((tree_vertices[8] - glob_ref_point[2]), 2));
+    /* The closer the reference point is to the intersection, the bigger is the scaling factor. */
+    *scaling_factor = dist_ref / dist_intersection;
```

Comment:
You can definetly remove the code duplication caused by the switch statements. Use switch statements only, when really needed.
Furthermore, `x*x` is always faster than `pow(x, 2)`. Even though it looks ugly, it is significantly faster